packages/react-core/src/hooks/useDropZone.ts

```diff
@@ -81,11 +81,54 @@ export default function useDropZone({
     setDragState(rejectedFiles.length > 0 ? 'reject' : 'accept');
   };
 
-  const onDrop = (event: React.DragEvent<HTMLDivElement>) => {
+  const onDrop = async (event: React.DragEvent<HTMLDivElement>) => {
```

Comment:
Can clear [the issue here](https://github.com/aws-amplify/amplify-ui/pull/6056/files#file-packages-react-src-primitives-dropzone-dropzone-tsx-L37) (and in future UI consumers) by dropping the `async`/`await` here in favor of async callbacks